autosklearn/pipeline/components/data_preprocessing/feature_type.py

```diff
@@ -40,31 +41,31 @@ def __init__(self, config=None, pipeline=None, dataset_properties=None, include=
         self.exclude = exclude
         self.random_state = random_state
         self.init_params = init_params
-        self.categorical_features = categorical_features
+        self.categorical_features = None
+        if categorical_features is not None:
+            self.categorical_features = categorical_features.tolist()
         self.force_sparse_output = force_sparse_output
 
         # The pipeline that will be applied to the categorical features (i.e. columns)
         # of the dataset
         # Configuration of the data-preprocessor is different from the configuration of
         # the categorical pipeline. Hence, force to None
         # It is actually the call to set_hyperparameter who properly sets this argument
-        # TODO: Extract the child configuration space from the datapreprocessor to the
+        # TODO: Extract the child configuration space from the FeatTypeSplit to the
         # pipeline if needed
         self.categ_ppl = CategoricalPreprocessingPipeline(
             config=None, steps=pipeline, dataset_properties=dataset_properties,
-            include=include, exclude=exclude, random_state=random_state,
-            init_params=init_params)
+            random_state=random_state, init_params=init_params)
```

Comment:
Sorry for the trouble, could you elaborate why include and exclude are removed here?

Reply:
Earlier there no configuration for data preprocessing step. Thus None was used for include/exclude here. But as now the step is configurable, we need to carefully handle the step. It's kinda parallel pipelines (which have configurable steps such as rescaling, cat encoding) within pipeline. For now I added it back, and removed include and exclude from where they were passed here. 
Please let me know if it is required to have include and exclude for num/cat pipeline steps. We may need to think how the schema of include/exclude option look like for complex pipeline structures. 
Example for current implemented schema for include/exclude
```
{
    'classifier' : ['sgd','lda'],
    'feature_preprocessor': ['pca', 'kernel_pca']
}
```
What do you think of below schema structure for complex pipelines?
```
{
        'classifier': ['sgd', 'lda'],
        'feature_preprocessor': ['pca', 'kernel_pca'],
        'data_preprocessor': {
            'numerical_transformer': {
                'rescaling': ['minmax', 'normalize']
            },
            'categorical_transformer': {
                'category_coalescence': ['no_coalescense'],
                'categorical_encoding': ['no_encoding']
            }
        }
    }
```
Please let me know if this makes sense I will make changes accordingly.

Reply:
Thanks for your reply. So the question we want to answer here is if someone wants to use for instance only `'minmax'` scaling. But what happens if there are no numerical columns? We have to take into account multiple scenarios.

To my understanding, if someone wants to use only `minmax` as a data-preprocessing he/she can do so via:
```
autosklearn.pipeline.components.data_preprocessing.add_preprocessor(<custom preprocessing>)
```

So adding support for these custom scenarios might complicate the code and there is a work-around solution for these custom cases. In order words, I think your change gives enough flexibility to the users as it is. What do you think?

Reply:
Agreed, it will complicate code. We can go ahead with current implementation. Let me know if you have any other feedback.